modules/tinymce/src/plugins/table/main/ts/ui/RowDialog.ts

```diff
@@ -52,79 +52,74 @@ const switchRowType = (dom: DOMUtils, rowElm: HTMLElement, toType: string) => {
   }
 };
 
-const updateAdvancedProps = (modifier: DomModifier, data: RowData): void => {
+const updateSimpleProps = (modifier: DomModifier, data: RowData) => {
+  modifier.setAttrib('scope', data.scope);
+  modifier.setAttrib('class', data.class);
+  modifier.setStyle('height', Util.addSizeSuffix(data.height));
+};
+
+const updateAdvancedProps = (modifier: DomModifier, data: RowData) => {
   modifier.setStyle('background-color', data.backgroundcolor);
   modifier.setStyle('border-color', data.bordercolor);
   modifier.setStyle('border-style', data.borderstyle);
 };
 
-const onSubmitRowForm = (editor: Editor, rows: HTMLElement[], oldData: RowData, api) => {
+const applyRowData = (editor: Editor, rows: HTMLElement[], oldData: RowData, data: RowData) => {
   const dom = editor.dom;
+  const isSingleRow = rows.length === 1;
 
-  const data: RowData = api.getData();
-  api.close();
+  Arr.each(rows, (rowElm) => {
+    // Switch row type
+    if (data.type !== rowElm.parentNode.nodeName.toLowerCase()) {
+      switchRowType(editor.dom, rowElm, data.type);
+    }
 
-  // When selection length is 1, allow things to be turned off/cleared
-  const createModifier: (dom, node: Node) => DomModifier = rows.length === 1 ? DomModifier.normal : DomModifier.ifTruthy;
+    const modifier = isSingleRow ? DomModifier.normal(dom, rowElm) : DomModifier.ifTruthy(dom, rowElm);
 
-  editor.undoManager.transact(() => {
-    Tools.each(rows, (rowElm) => {
+    updateSimpleProps(modifier, data);
 
-      // Switch row type
-      if (data.type !== rowElm.parentNode.nodeName.toLowerCase()) {
-        switchRowType(editor.dom, rowElm, data.type);
-      }
-
-      const modifier = createModifier(dom, rowElm);
+    if (hasAdvancedRowTab(editor)) {
+      updateAdvancedProps(modifier, data);
+    }
 
-      modifier.setAttrib('scope', data.scope);
-      modifier.setAttrib('class', data.class);
-      modifier.setStyle('height', Util.addSizeSuffix(data.height));
+    if (data.align !== oldData.align) {
+      Styles.unApplyAlign(editor, rowElm);
+      Styles.applyAlign(editor, rowElm, data.align);
+    }
+  });
+};
 
-      if (hasAdvancedRowTab(editor)) {
-        updateAdvancedProps(modifier, data);
-      }
+const onSubmitRowForm = (editor: Editor, rows: HTMLElement[], oldData: RowData, api) => {
+  const data: RowData = api.getData();
+  api.close();
 
-      if (data.align !== oldData.align) {
-        Styles.unApplyAlign(editor, rowElm);
-        Styles.applyAlign(editor, rowElm, data.align);
-      }
-    });
+  editor.undoManager.transact(() => {
+    applyRowData(editor, rows, oldData, data);
     editor.focus();
   });
 };
 
-const open = (editor: Editor) => {
-  const dom = editor.dom;
-  let tableElm, cellElm, rowElm;
-  const rows = [];
-
-  tableElm = dom.getParent(editor.selection.getStart(), 'table');
+const getRowsFromSelection = (editor: Editor): HTMLTableRowElement[] => {
```

Comment:
I'm thinking this and getCellsFromSelection will probably end up in another file eventually, because they're probably helpful in a few places, including API stuff I'm looking at doing for headers.